libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -485,18 +496,24 @@ Optional<Class<?>> findRequestingClass(Stream<Class<?>> classes) {
      * @return true if permission is granted regardless of the entitlement
      */
     private static boolean isTriviallyAllowed(Class<?> requestingClass) {
-        if (logger.isTraceEnabled()) {
-            logger.trace("Stack trace for upcoming trivially-allowed check", new Exception());
-        }
         if (requestingClass == null) {
-            logger.debug("Entitlement trivially allowed: no caller frames outside the entitlement library");
+            // This is unexpected and unusual; we probably want to log at the info level
+            logger.info("Entitlement trivially allowed: no caller frames outside the entitlement library", new Exception("STACK TRACE"));
```

Comment:
Should this be an error, or at least a warning?  If this is unexpected why not throw an AssertionError? This probably deserves to be in a separate PR, it seems unrelated to adding thread entitlements.